spec/integration/transaction_spec.rb

```diff
@@ -3,7 +3,13 @@
     Dry.Transaction(container: container) do
       map :process
       step :verify
-      try :validate, catch: Test::NotValidError
+      try :validate, with: -> input {
+        if input[:email].nil?
+          raise(container[:invalid_error], "email required")
+        else
+          input
+        end
+      }, catch: Test::NotValidError
```

Comment:
It's because we allow blocks passed to the step definitions to be passed right along to the step adapters themselves (as of https://github.com/dry-rb/dry-transaction/pull/36).